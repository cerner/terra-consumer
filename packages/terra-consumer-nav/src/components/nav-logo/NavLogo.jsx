import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Card from '../card/Card';
import styles from './NavLogo.scss';

const propTypes = {
  /**
   *  The url of the logo to be shown.
   */
  url: PropTypes.string.isRequired,
  /**
   *  The alternate text that is read by screen readers or displayed if the image fails to load.
   */
  altText: PropTypes.string,
  /**
   * Whether or not the logo should be placed on top of a white card.
   */
  isCard: PropTypes.bool,
};

const defaultProps = {
  altText: 'Client Logo',
  isCard: false,
};

const cx = classNames.bind(styles);

const NavLogo = ({
  url,
  altText,
  isCard,
  ...customProps
}) => {
  const image = <img className={cx('img')} src={url} alt={altText} />;
  const domNode = isCard ? Card : 'div';
  const logoClassNames = cx(
    'logo-container',
    customProps.className,
  );

  return React.createElement(domNode, { ...customProps, className: logoClassNames }, image);
};

NavLogo.propTypes = propTypes;
NavLogo.defaultProps = defaultProps;

export default NavLogo;
