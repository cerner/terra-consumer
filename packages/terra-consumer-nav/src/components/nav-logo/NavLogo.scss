@import '../../variables';

:local {
  .logo-container {
    margin: 0 25px 30px;
    max-height: 170px;
    max-width: calc(100% - 50px});
    overflow: hidden;
    text-align: center;
    @media screen and (max-width: $terra-medium-breakpoint) {
      margin: 40px 25px;
    }
  }

  .img {
    color: var(--terra-consumer-logo-text-color, #3d3d3d);
    display: inline-block;
    font-size: 30px;
    font-weight: normal;
    margin: auto;
    max-height: 120px;
    max-width: 100%;
  }
}
