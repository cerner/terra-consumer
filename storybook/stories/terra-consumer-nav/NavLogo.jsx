import React from 'react';
import NavLogo from 'terra-consumer-nav/lib/components/nav-logo/NavLogo';

const logoWithOutURL = {
  url: '',
  altText: 'Nike',
  isCard: true,
};

const logoWithURL = {
  url: 'http://photos.prnewswire.com/prnfull/20140210/DC61972LOGO',
  altText: 'Placeholder logo',
  isCard: true,
};

const logoWithURL2 = {
  url: 'http://therunnersflat.com/wp-content/uploads/2014/05/RF-Mark-Revs-V1.png',
  altText: 'Placeholder logo',
  isCard: false,
};

const styles = Object.assign({}, { maxWidth: '250px' });


export default () => (
  <div>
    <h2>Client wants text as logo </h2>
    <div>
      <NavLogo {...logoWithOutURL} style={styles} />
    </div>
    <h2>Client has logo</h2>
    <div>
      <NavLogo {...logoWithURL} style={styles} />
    </div>
    <div>
      <NavLogo {...logoWithURL2} style={styles} />
    </div>
  </div>
);
