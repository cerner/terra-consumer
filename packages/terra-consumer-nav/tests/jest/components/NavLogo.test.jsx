import React from 'react';
import NavLogo from '../../../src/components/nav-logo/NavLogo';

const testData = {
  url: '#testPath',
  altText: 'testIcon',
};

describe('Nav Logo', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<NavLogo {...testData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Card instead of a div when specified', () => {
    const wrapper = shallow(<NavLogo {...testData} isCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply custom classes', () => {
    const wrapper = shallow(<NavLogo {...testData} className="test-class" />);
    expect(wrapper).toMatchSnapshot();
  });
});
