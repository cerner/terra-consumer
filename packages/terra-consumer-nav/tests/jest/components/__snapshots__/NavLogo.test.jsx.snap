// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Logo should apply custom classes 1`] = `
<div
  className="logo-container test-class"
>
  <img
    alt="testIcon"
    className="img"
    src="#testPath"
  />
</div>
`;

exports[`Nav Logo should render a Card instead of a div when specified 1`] = `
<Card
  className="logo-container"
>
  <CardBody
    hasPaddingHorizontal={true}
    hasPaddingVertical={true}
  >
     
    <img
      alt="testIcon"
      className="img"
      src="#testPath"
    />
     
  </CardBody>
</Card>
`;

exports[`Nav Logo should render a default component 1`] = `
<div
  className="logo-container"
>
  <img
    alt="testIcon"
    className="img"
    src="#testPath"
  />
</div>
`;

exports[`Nav Logo should render a div regardless of isCard when no url is provided 1`] = `
<div
  className="logo-container"
>
  <p
    className="img"
  >
    test
  </p>
</div>
`;
